### DISCUSSION TITLE
RFEM 6 - Nodal Supports - Translational and Rotational spring constants

### DISCUSSION THREAD

**Discussion by EfrainDuran** [Dlubal-Software/Dlubal_CSharp_Client#62] [Q&A] [answered]
Hello can you help me with nodal supports, I'm trying to set to true the checkboxes on Translational like this:
![image](https://github.com/user-attachments/assets/951a43d0-e7fa-4a07-904b-c067735ad1e4)

on RFEM5 i used to set the value to -1 and that worked. RFEM 6 doesn't accept negative values.

RFEM5 example:

	NodalSupport ns = new NodalSupport
	{
		ID = "NodalSupport",
		NodeList = Number.ToString(),
		ReferenceSystem = ReferenceSystemType.UserDefinedSystemType,
		No = n,
		UserDefinedReferenceSystem = new UserDefinedAxisSystem()
		{
			Type = UserDefinedAxisSystemType.SameAsLineType,
			ObjectNo = thisLine.No
		},
		SupportConstantX = 0,
		SupportConstantY = -1,
		SupportConstantZ = -1,
		RestraintConstantX = 339,
		RestraintConstantY = 0,
		RestraintConstantZ = 0
	};

**Comment by NathanMSY** [comment, marked as answer]
If you use the "get_nodal_support" request to get an existing nodal support form an RFEM model, you can probably see how to set a rigid support.
I quickly used Boomerang to make the SOAP call and the XML coming back looks like this:

`<rotational_restraint_x>200000</rotational_restraint_x>`
`<rotational_restraint_y>0</rotational_restraint_y>`
`<rotational_restraint_z>INF</rotational_restraint_z>`

INF is rigid
